orchagent/routeorch.cpp

```diff
@@ -1794,10 +1899,18 @@ bool RouteOrch::removeRoutePost(const RouteBulkContext& ctx)
             && m_syncdNextHopGroups[it_route->second].ref_count == 0)
         {
             m_bulkNhgReducedRefCnt.emplace(it_route->second);
-        } else if (ol_nextHops.is_overlay_nexthop()){
+        }
+        else if (ol_nextHops.is_overlay_nexthop())
+        {
             SWSS_LOG_NOTICE("Remove overlay Nexthop %s", ol_nextHops.to_string().c_str());
             removeOverlayNextHops(vrf_id, ol_nextHops);
         }
+        else if (ol_nextHops.getSize() == 1)
```

Comment:
it is better to also check if the nexthop is pointing to the muxcable port or not.